app/forms/steps/income/armed_forces_form.rb

```diff
@@ -0,0 +1,47 @@
+module Steps
+  module Income
+    class ArmedForcesForm < Steps::BaseFormObject
+      include Steps::HasOneAssociation
+
+      attr_accessor :subject
+
+      has_one_association :income
+
+      attribute :in_armed_forces, :value_object, source: YesNoAnswer
+
+      validate :in_armed_forces_choice
```

Comment:
I think this would handle it, so there would be no need for the `in_armed_forces_choice` method 

```suggestion
      validates :in_armed_forces, inclusion: { in: :choices }
```